lxd/storage/drivers/driver_common.go

```diff
@@ -434,7 +435,14 @@ func (d *common) RenameVolumeSnapshot(snapVol Volume, newSnapshotName string, op
 
 // ValidateBucket validates the supplied bucket name.
 func (d *common) ValidateBucket(bucket Volume) error {
-	match, err := regexp.MatchString(`^[a-z0-9][\-\.a-z0-9]{2,62}$`, bucket.name)
+	bucketName := bucket.name
+
+	// Trim project name prefix if applicable
+	if strings.Contains(bucketName, "_") {
```

Comment:
Given that we are now always supposed to be prefixing it with project name, I think we should actually return an error if the bucketName is not containing an underscore, we can use strings.Cut() for this I think to extract the bucket name to validate.

Reply:
Oh we can just use StorageVolumeParts directly as you have done. Just without the Contains check

Reply:
And we should fail if project name is empty.